examples/streams/src/main.rs

```diff
@@ -124,3 +151,83 @@ async fn stream_transactions(
     }
     Ok(())
 }
+
+/// Subscribes to receipts related to a specific contract, effectively listening to contract events.
+///
+/// This function creates a stream that subscribes to various types of receipts (except ScriptResult
+/// and MessageOut) that are associated with the specified contract ID. It's a way to monitor
+/// contract-related events such as calls, returns, logs, transfers, mints, and burns.
+///
+/// The function filters the receipts to ensure they match the given contract ID before processing them.
+/// This approach allows for efficient monitoring of contract activities without the need to process
+/// irrelevant receipts.
+///
+/// # Arguments
+///
+/// * `client` - A reference to the NATS client used for streaming.
+/// * `contract_id` - The ID of the contract to monitor.
+///
+/// # Returns
+///
+/// Returns a Result which is Ok if the streaming completes successfully, or an Error if there are any issues.
+async fn stream_contract(
+    client: &Client,
+    contract_id: ContractId,
+) -> anyhow::Result<()> {
+    let mut receipt_stream = fuel_streams::Stream::<Receipt>::new(client).await;
+
+    receipt_stream.with_filter(
+        ReceiptsBurnSubject::new().with_contract_id(Some(contract_id.into())),
+    );
+    receipt_stream.with_filter(
+        ReceiptsCallSubject::new().with_from(Some(contract_id.into())),
+    );
```

Comment:
This might be better from a readability standpoint:

```suggestion
    let mut receipt_stream = fuel_streams::Stream::<Receipt>::new(client).await;
    let receipt_stream = receipt_stream.with_filter(
        ReceiptsBurnSubject::new().with_contract_id(Some(contract_id.into())),
    ).with_filter(
        ReceiptsCallSubject::new().with_from(Some(contract_id.into())),
    ).
    // other with_filters come here
```